File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.js
function solution(polynomial) {
    var answer = '';
    let xSum = 0;
    let numSum =0;
    let poly = polynomial.split(" + ");
    
    for(let i=0; i<poly.length; i++){
        if(poly[i].includes("x")){
            let temp = poly[i].split("x")[0];
            
            if(temp === "") xSum += 1
            else xSum += parseInt(temp);
        }else{
            numSum += parseInt(poly[i]);
        }
        }
    
    if(numSum === 0){
        answer = xSum === 1 ? "x" : `${xSum}x`;    
    }else if(xSum === 0){
        answer = `${numSum}`;    
    }else{
        answer = xSum === 1 ? `x + ${numSum}` : `${xSum}x + ${numSum}`;
    }  
    return answer;
}

File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    xSum=0
    numSum=0
    
    for p in polynomial.split(" + "):
        if p.endswith("x"):
            if p == "x":
                xSum += 1
            else:
                xSum += int(p[:-1])
        else:
            numSum += int(p)
    if xSum == 0:
        return str(numSum)
    elif numSum == 0:
        return 'x' if xSum == 1 else f"{xSum}x"
    else:
        return f"x + {numSum}" if xSum == 1 else f"{xSum}x + {numSum}"
        
